More prep for packaging

edenchazard opened the pull request:
Adds the basic typescript setup and frontend functionality in package form.

willpower232 reviewed (changes requested):
I'm not sure why you changed the controller to make it a separate function, it's all cool but definitely going to conflict now :sweat_smile:  I added more config too to do the keys of `$values`

You wanna have a go at writing out the readme? The public in public const in the enum means the minimum PHP is 8.2 but I added back to laravel 9 in the tests because they're spectacularly vanilla.

The developer just has to `\SynergiTech\MagicEnums\MagicEnumsRouteFacade::enumsController();` somewhere in their routes file and give the plugin the same route and then the magic can happen.

Definitely open to whatever other tweaks, at least the tests now pass in main so you can see where things go funky.

edenchazard commented:
> I'm not sure why you changed the controller to make it a separate function, it's all cool but definitely going to conflict now 😅 I added more config too to do the keys of $values

Honestly, I think I was attempting something and then got distracted. I've just reverted these changes for now.

> You wanna have a go at writing out the readme? 

Sure.

willpower232 reviewed (approved):
lets gooooo
